sympy/solvers/solveset.py

```diff
@@ -978,6 +985,353 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _expo_solver(f):
+    r"""
+    Helper function for solving exponential equations.
+
+    Exponential equations are the type of equations which has a
```

Comment:
"A sum of (currently) at most 2 terms with one or both of them having a power with a symbol-dependent exponent."

(That `exp` is a function is an internal detail that I wouldn't bother differentiating between in the docstring, but would make the second example below involve `exp`, e.g. change the `8^` to `exp`.